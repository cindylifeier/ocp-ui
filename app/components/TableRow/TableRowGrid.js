import { Grid } from 'styled-css-grid';

const TableRowGrid = Grid.extend`
  border-bottom: 1px outset rgb(51, 51, 51);

  &:nth-child(odd) {
    background-color: white;
  }

  &:first-child {
    background-color: white;
    font-size: 20px;
  }

  &:hover {
    background-color: #dce4ef;
    cursor: pointer;
  }

  &:focus {
    background-color: #93b9ea;
  }
`;

TableRowGrid.propTypes = {};

export default TableRowGrid;
