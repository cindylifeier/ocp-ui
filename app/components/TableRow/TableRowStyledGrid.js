import { Grid } from 'styled-css-grid';

const TableRowStyledGrid = Grid.extend`
  border-bottom: 1px outset rgb(51, 51, 51);

  &:nth-child(odd) {
    background-color: #f2f2f2;
  }

  &:first-child {
    background-color: #f2f2f2;
    font-size: 20px;
  }

  &:hover {
    background-color: #dce4ef;
    cursor: pointer;
  }
`;

TableRowStyledGrid.propTypes = {};

export default TableRowStyledGrid;
