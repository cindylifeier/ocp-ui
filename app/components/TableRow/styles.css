.rowGridContainer:nth-child(odd) {
  background-color: #f2f2f2;
}

.rowGridContainer:first-child {
  background-color: #f2f2f2;
  font-size: 20px;
}

.rowGridContainer:hover {
  background-color: #dce4ef;
  cursor: pointer;
}

.rowGridContainer {
  display: grid;
  grid-gap: 5px;
  border-bottom: 1px outset rgb(51, 51, 51);
}
