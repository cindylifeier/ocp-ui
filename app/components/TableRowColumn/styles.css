.cellGridItem {
  color: rgb(51, 51, 51);
  font-family: "Arial Bold", "Arial";
  font-style: normal;
  font-size: 12px;
  font-weight: 700;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-all;
  padding: 2px;
}
