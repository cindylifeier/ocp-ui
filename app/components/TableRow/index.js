/**
 *
 * TableRow
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

// import styled from 'styled-components';

function TableRow({ children }) {
  return (
    <div
      className={styles.rowGridContainer}
      style={{ gridTemplateColumns: `repeat(${React.Children.count(children)}, 1fr)` }}
    >
      {children}
    </div>
  );
}

TableRow.propTypes = {
  children: PropTypes.node,
};

export default TableRow;
