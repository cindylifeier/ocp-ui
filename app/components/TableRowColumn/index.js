/**
 *
 * TableRowColumn
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles.css';

// import styled from 'styled-components';

function TableRowColumn({ children }) {
  return (
    <div className={styles.cellGridItem}>
      {children}
    </div>
  );
}

TableRowColumn.propTypes = {
  children: PropTypes.node,
};

export default TableRowColumn;
