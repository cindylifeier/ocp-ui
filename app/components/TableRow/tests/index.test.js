import React from 'react';
import { shallow } from 'enzyme';

import TableRow from '../index';
import TableRowColumn from '../../TableRowColumn';

describe('<TableRow />', () => {
  it('should match snapshot', () => {
    // Act
    const renderedComponent = shallow(
      (<TableRow>
        <TableRowColumn>Row</TableRowColumn>
      </TableRow>)
    );

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
});
