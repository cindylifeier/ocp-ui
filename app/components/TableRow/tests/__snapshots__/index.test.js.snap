// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> shoud match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}
    style={
        Object {
            "gridTemplateColumns": "repeat(1, 1fr) 0.5fr",
          }
    }
>
    <TableRowColumn>
        Row
    </TableRowColumn>
</div>,
  "nodes": Array [
    <div
      className={undefined}
      style={
            Object {
                  "gridTemplateColumns": "repeat(1, 1fr) 0.5fr",
                }
      }
>
      <TableRowColumn>
            Row
      </TableRowColumn>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TableRow>
        <TableRowColumn>
                Row
        </TableRowColumn>
</TableRow>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <TableRowColumn>
            Row
</TableRowColumn>,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}
          style={
                    Object {
                              "gridTemplateColumns": "repeat(1, 1fr) 0.5fr",
                            }
          }
>
          <TableRowColumn>
                    Row
          </TableRowColumn>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}
          style={
                    Object {
                              "gridTemplateColumns": "repeat(1, 1fr) 0.5fr",
                            }
          }
>
          <TableRowColumn>
                    Row
          </TableRowColumn>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TableRow>
    <TableRowColumn>
        Row
    </TableRowColumn>
</TableRow>,
}
`;
